tests/integration/common/state.rs

```diff
@@ -295,26 +304,35 @@ impl<'a> StarknetStateBuilder<'a> {
     }
 
     /// Funds all accounts according to the test fee configuration.
-    fn fund_accounts(fee_config: &FeeConfig, dict_state_reader: &mut DictStateReader) {
-        let mut addresses: HashSet<ContractAddress> = Default::default();
+    fn fund_accounts(
+        fee_config: &FeeConfig,
+        dict_state_reader: &mut DictStateReader,
+        mut funds_per_address: HashMap<ContractAddress, Balance>,
+    ) {
         for address in dict_state_reader.address_to_class_hash.keys().chain(dict_state_reader.address_to_nonce.keys()) {
-            addresses.insert(*address);
+            if !funds_per_address.contains_key(address) {
+                funds_per_address.insert(address.clone(), fee_config.default_balance.clone());
```

Comment:
I think this would silently drop duplicates, which isn't much of a concern except that would be a real PITA to debug. A simple `log::warn` or `assert` could be appropriate